output.go

```diff
@@ -68,21 +97,80 @@ func (m *multiOutput) ClosePipe(proc *process) {
 	}
 }
 
-func (m *multiOutput) WriteLine(proc *process, p []byte) {
+func (m *multiOutput) textLogLine(proc *process, p []byte, style stringStyle) bytes.Buffer {
 	var buf bytes.Buffer
 
-	color := fmt.Sprintf("\033[1;%vm", proc.Color)
+	var colorCode int
+	if style == styleError {
+		colorCode = ColorErr
+	} else {
+		colorCode = proc.Color
+	}
+
+	color := fmt.Sprintf("\033[1;%vm", colorCode)
+
+	if m.ColorizeOutput {
+		buf.WriteString(color)
+	}
 
-	buf.WriteString(color)
 	buf.WriteString(proc.Name)
 
-	for buf.Len()-len(color) < m.maxNameLength {
-		buf.WriteByte(' ')
+	if m.ColorizeOutput {
+		for buf.Len()-len(color) < m.maxNameLength {
+			buf.WriteByte(' ')
+		}
+	} else {
+		for buf.Len() < m.maxNameLength {
+			buf.WriteByte(' ')
+		}
+	}
+
+	if m.ColorizeOutput {
+		buf.WriteString("\033[0m | ")
+	} else {
+		buf.WriteString(" | ")
+	}
+
+	if style == styleBold && m.ColorizeOutput {
+		buf.WriteString("\033[1m")
 	}
 
-	buf.WriteString("\033[0m | ")
 	buf.Write(p)
+
+	if style == styleBold {
+		buf.WriteString("\033[0m")
+	}
+
 	buf.WriteByte('\n')
+	return buf
+}
+
+func (m *multiOutput) jsonLogLine(proc *process, p []byte) bytes.Buffer {
+	var buf bytes.Buffer
+
+	data := jsonlog{
+		Process: proc.Name,
+		Line: string(p),
+	}
+
+	line, _ := json.Marshal(&data)
+	buf.Write(line)
+	buf.WriteByte('\n')
+	return buf
+}
+
+func (m *multiOutput) WriteLine(proc *process, p []byte, style stringStyle) {
+
+	var buf bytes.Buffer
+
+	switch m.LogFormat {
+	case logFormatText:
+		buf = m.textLogLine(proc, p, style)
+	case logFormatJSON:
+		buf = m.jsonLogLine(proc, p)
+	default:
+		panic("BUG: Log Format was invalid!")
```

Comment:
It would be better to check the log format before you run any process. This is not a bug, this is an invalid input from a user.
Also `panic` isn't a good idea here, an end user doesn't want to see a backtrace. Use `fatal` for such cases.